support: Introduce editorconfig

nderjung opened the pull request:
EditorConfig [specifies](https://editorconfig.org) how an IDE configures generic styling for files for a project.  This is done by introducing a top-level `.editorconfig` file within the repository. This PR sets the Unikraft standards using EditorConfig:

 * indentation using tabs with spacing of 8 characters;
 * max line lenfth of 75 characters for git commit messages;
 * max line length of 80 characters for standard files;
 * adding a new line to the end of files; and,
 * trimming of whitespace.

I'm more of a "spaces" guy myself, so my editor is set up that way.  With this file, and when working on Unikraft, I don't have to think twice about what happens when hitting the "tab" key.

nderjung committed:
support: Introduce editorconfig

EditorConfig[0] specifies how an IDE configures generic styling for files
for a project.  This is done by introducing a top-level `.editorconfig`
file within the repository.  This commit sets the Unikraft standards using
EditorConfig:

 * indentation using tabs with spacing of 8 characters;
 * max line lenfth of 75 characters for git commit messages;
 * max line length of 80 characters for standard files;
 * adding a new line to the end of files; and,
 * trimming of whitespace.

[0]: https://editorconfig.org

Signed-off-by: Alexander Jung <a.jung@lancs.ac.uk>

squash

skuenzer commented:
Nice file, I would also add
```
trim_trailing_whitespace = true
```

I saw there is also an option to limit to 80 columns but I am unsure how brutally this is enforced. Similar to the Linux kernel folks, we have some exceptions, like print format arguments that could be longer than 80 columns.

nderjung commented:
Hey @skuenzer, I've updated the file with your recommendations :)

skuenzer reviewed (changes requested):
Is there a typo in the commit subject?
I would choose `support` as selector and shorten the title:
`support: Introduce editorconfig`

In the commit message body I would add a short explanation what editorconfig is.

nderjung commented:
I've updated this PR and added an extra rule to address the latest checkpatch:
```
WARNING:COMMIT_LOG_LONG_LINE: Possible unwrapped commit description (prefer a maximum 75 chars per line)
```
by setting `max_line_length = 75` for anything within the `.git` directory.

skuenzer reviewed (approved):
Cool, looks good to me! Thanks.

Reviewed-by: Simon Kuenzer <simon.kuenzer@neclab.eu>
